app/models/sample.rb

```diff
@@ -246,8 +251,12 @@ def fastqs_folder_files
   end
 
   def initiate_input_file_upload
-    return unless input_files.first.source_type == InputFile::SOURCE_TYPE_S3
-    Resque.enqueue(InitiateS3Cp, id)
+    if !input_files.empty? && input_files.first.source_type == InputFile::SOURCE_TYPE_S3
+      Resque.enqueue(InitiateS3Cp, id)
+    end
+    if basespace_upload?
```

Comment:
should this be `elif` ? and it seems like there should by symmetry with `InputFile::SOURCE_TYPE_S3` ... why do we branch on `basespace_dataset_id.present?` and not `InputFile::SOURCE_TYPE_BASESPACE` ?

Reply:
It's a limitation of the basespace API.

I added this comment to the code:
```
    # The reason why we don't check input_files.first.source_type == InputFile::SOURCE_TYPE_BASESPACE here is:
    # We don't yet have the input files at this point.
    # The basespace API doesn't let us query for all input files for a project.
    # We can only query all the datasets (and get the dataset IDs).
    # We then need to query each dataset individually (with a separate API call) to get the input files.
    # We do this query inside the TransferBasespaceFiles resque task.
```